src/app/paint/mod.rs

```diff
@@ -1006,7 +1052,5 @@ impl AppState for State {
       }
    }
 
-   fn exit(self: Box<Self>) {
-      tokio::spawn(self.xcoder.shutdown());
-   }
+   fn exit(self: Box<Self>) {}
 }
```

Comment:
You seem to have removed code responsible for shutting down the transcoding thread cleanly. If we have loose threads hanging around at the time of pressing the Close button, the app will wait some time until forcibly shutting them down, which causes a noticable delay when closing (and a log message is emitted signalling this.)

Could you come up with some sort of shutdown mechanism for the encoding/decoding threads to avoid this?

Reply:
I specifically made our `main` block for two seconds if there are any tasks that don't manage to shut down in time: https://github.com/liquidev/netcanv/blob/master/src/main.rs#L326

This was to ensure that any sockets get shut down properly, otherwise the connection would be torn down abruptly which would result in an error on the relay.

Reply:
Ah, I misunderstood what the code does. I'm reading it again now and the tasks that are spawned are short-lived and there is no more main decoding loop, is that correct?